libnymea-core/zwave/zwavenodeimplementation.cpp

```diff
@@ -0,0 +1,341 @@
+        emit nodeChanged();
+    }
+}
+
+QString ZWaveNodeImplementation::manufacturerName() const
+{
+    return m_manufacturerName;
+}
+
+void ZWaveNodeImplementation::setManufacturerName(const QString &manufacturerName)
+{
+    if (m_manufacturerName != manufacturerName) {
+        m_manufacturerName = manufacturerName;
+        emit manufacturerNameChanged();
+        emit nodeChanged();
+    }
+}
+
+QString ZWaveNodeImplementation::name() const
+{
+    return m_name;
+}
+
+void ZWaveNodeImplementation::setName(const QString &name)
+{
+    if (m_name != name) {
+        m_name = name;
+        emit nameChanged();
+        emit nodeChanged();
+    }
+}
+
+quint16 ZWaveNodeImplementation::productId() const
+{
+    return m_productId;
+}
+
+void ZWaveNodeImplementation::setProductId(quint16 productId)
+{
+    if (m_productId != productId) {
+        m_productId = productId;
+        emit productIdChanged();
+        emit nodeChanged();
+    }
+}
+
+QString ZWaveNodeImplementation::productName() const
+{
+    return m_productName;
+}
+
+void ZWaveNodeImplementation::setProductName(const QString &productName)
+{
+    if (m_productName != productName) {
+        m_productName = productName;
+        emit productNameChanged();
+        emit nodeChanged();
+    }
+}
+
+quint16 ZWaveNodeImplementation::productType() const
+{
+    return m_productType;
+}
+
+void ZWaveNodeImplementation::setProductType(quint16 productType)
+{
+    if (m_productType != productType) {
+        m_productType = productType;
+        emit productTypeChanged();
+        emit nodeChanged();
+    }
+}
+
+quint8 ZWaveNodeImplementation::version() const
+{
+    return m_version;
+}
+
+void ZWaveNodeImplementation::setVersion(quint8 version)
+{
+    if (m_version != version) {
+        m_version = version;
+        emit versionChanged();
+        emit nodeChanged();
+    }
+}
+
+bool ZWaveNodeImplementation::isZWavePlus() const
+{
+    return m_isZWavePlus;
+}
+
+void ZWaveNodeImplementation::setIsZWavePlus(bool isZWavePlus)
+{
+    if (m_isZWavePlus != isZWavePlus) {
+        m_isZWavePlus = isZWavePlus;
+        emit isZWavePlusChanged();
+        emit nodeChanged();
+    }
+}
+
+bool ZWaveNodeImplementation::isSecure() const
+{
+    return m_isSecure;
+}
+
+void ZWaveNodeImplementation::setIsSecure(bool isSecure)
+{
+    if (m_isSecure != isSecure) {
+        m_isSecure = isSecure;
+        emit isSecureChanged();
+        emit nodeChanged();
+    }
+}
+
+bool ZWaveNodeImplementation::isBeaming() const
+{
+    return m_isBeaming;
+}
+
+void ZWaveNodeImplementation::setIsBeaming(bool isBeaming)
+{
+    if (m_isBeaming != isBeaming) {
+        m_isBeaming = isBeaming;
+        emit isBeamingChanged();
+        emit nodeChanged();
+    }
+}
+
+void ZWaveNodeImplementation::updateValue(const ZWaveValue &value)
+{
+    if (m_values.contains(value.id())) {
+        m_values[value.id()] = value;
+        emit valueChanged(value);
+//        emit nodeChanged();
```

Comment:
No node changed if the value changes?

Reply:
Also for value removed?

Reply:
Right. I decided to not emit nodeChanged() on value changes. This is like attribute changed in zigbe... nodeChanged will only be emitted for node descriptor stuff changing... If interested in values, valueChanged is way to go.